app/repositories/slick/implementations/SlickChatsRepository.scala

```diff
@@ -236,19 +239,43 @@ class SlickChatsRepository @Inject() (db: Database)(implicit executionContext: E
         case (chatID, subject, fromAddress) => Some(CreateChatDTO(
           Some(chatID),
           Some(subject),
-          CreateEmailDTO(
+          UpsertEmailDTO(
             emailId = Some(emailId),
-            from = fromAddress,
+            from = Some(fromAddress),
             to = createEmailDTO.to,
             bcc = createEmailDTO.bcc,
             cc = createEmailDTO.cc,
             body = createEmailDTO.body,
-            date = Some(date))))
+            date = Some(date),
+            sent = Some(false))))
       }
       case None => None
     }
   }
 
+  def patchEmail(upsertEmailDTO: UpsertEmailDTO, chatId: String, emailId: String, userId: String): Future[Option[Email]] = {
+    val updateAndSendEmail = for {
+      updatedReceiversAddresses <- updateEmail(upsertEmailDTO, chatId, emailId, userId)
+
+      sendEmail <- DBIO.sequenceOption(
+        updatedReceiversAddresses.map(receiversAddresses =>
+          if (upsertEmailDTO.sent.getOrElse(false))
+            sendEmail(userId, chatId, emailId, receiversAddresses)
+          else DBIO.successful(0)))
+
+    } yield sendEmail
+
+    for {
+      optionPatch <- db.run(updateAndSendEmail.transactionally)
+      email <- db.run(getEmailAction(userId, emailId))
+    } yield optionPatch.flatMap(_ => email.headOption)
+  }
+
+  def moveChatToTrash(chatId: String, userId: String): Future[Boolean] = {
+    db.run(UserChatsTable.moveChatToTrash(userId, chatId))
```

Comment:
This moveChatToTrash method already exists. And has already properly separated db.runs from DBIOAction